bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/AdaptiveImageServlet.java

```diff
@@ -606,6 +635,29 @@ private EnhancedRendition filter(@NotNull EnhancedRendition rendition) throws IO
         throw new IOException(String.format("Cannot process rendition %s due to size %s", rendition.getName(), rendition.getDimension()));
     }
 
+    private void streamOrConvert(@NotNull SlingHttpServletResponse response, @NotNull EnhancedRendition rendition, @NotNull String imageType,
+                                 String imageName, int resizeWidth, double quality) throws IOException {
+        Dimension dimension = rendition.getDimension();
+        if (rendition.getMimeType().equals(imageType)) {
+            LOGGER.debug("Found rendition {}/{} has a width of {}px and does not require a resize for requested width of {}px",
+                    rendition.getAsset().getPath(), rendition.getName(), dimension != null ? dimension.getWidth() : null, resizeWidth);
+            stream(response, rendition.getStream(), imageType, imageName);
```

Comment:
it would be better to use try..with...resources on the stream and close it where it has been opened (instead of closing it in the method stream (to which is is passed as parameter)